addons/sale_timesheet/models/sale_order.py

```diff
@@ -131,11 +136,14 @@ def action_view_timesheet(self):
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
+    task_id = fields.Many2one('project.task', 'Task')
+
     @api.model
     def create(self, values):
         line = super(SaleOrderLine, self).create(values)
         if line.state == 'sale' and not line.order_id.project_id and line.product_id.track_service in ['timesheet', 'task']:
```

Comment:
Shouldn't this call also `_is_task` method? If not, you won't have the same criteria (if modified by inherited modules) when confirming that when adding an extra line.

Reply:
Please attend this comment too